### ISSUE TITLE
[4.x] Issue on frontend login with MFA enabled

### ISSUE THREAD

**Issue by dautrich** [joomla/joomla-cms#40428]
### Steps to reproduce the issue

1. Create a user with MFA via TOTP enabled
2. Login to the frontend with username and password
![grafik](https://user-images.githubusercontent.com/7933215/233174046-7bb3dee3-2219-4dfc-a661-c06e41bf6e29.png)
3. Watch the next screen, where you have to enter your verification code

### Expected result
You will not see an alert message "You have been logged in" at this point.
This message should appear after you successfully entered your verification code, on the next screen.

### Actual result
You see an alert message "You have been logged in", although the login process is not finished yet.
![grafik](https://user-images.githubusercontent.com/7933215/233176999-e4f474bf-f0a7-443b-bd78-bda0eeda3640.png)

### System information (as much as possible)

[systeminfo-2023-04-19T21 08 46+02 00.txt](https://github.com/joomla/joomla-cms/files/11276735/systeminfo-2023-04-19T21.08.46%2B02.00.txt)

### Additional comments
I didn't test with other MFA methods (i.e. WebAuthn), but I suspect that the bug shows up as well.

**Comment by dautrich** [comment]
@richard67 The button might better be labeled "Cancel", in case you decide not to log in.

**Comment by richard67** [comment]
Well we also could use „f…ck, I can‘t find my smartphone with the authenticator app right now“, but that is a bit long, so „Cancel“ is fine :-)

**Comment by dautrich** [comment]
In my opinion, one should not see any content with an access level of "Registered", before the login process (including MFA) is completed. If I am right, the main menu should not appear in the form for 2FA, because the menu alone might disclose confidential information. Even submenus are visible:

![grafik](https://user-images.githubusercontent.com/7933215/233326632-791a1c38-0fa6-495a-bbd4-52c9b38843d6.png)

**Comment by brianteeman** [comment]
That is why there are settings for which modules are available 

![image](https://user-images.githubusercontent.com/1296369/233333905-b27f0089-1cd6-4b27-bc25-95e4d35907d8.png)

**Comment by dautrich** [comment]
> @dautrich Silly question: Can this issue also be reproduced with a 4.2.9, or does it really need the 4.3.0?

Yes, it is the same in 4.2.9.

**Comment by sandewt** [comment]
Suggestion: most simple (temporary) solution is, or in case MFA is enabled.

![Schermafbeelding 2023-04-20 160547](https://user-images.githubusercontent.com/9271775/233392094-b05427eb-84b6-48a8-b2db-7668c57719b5.png)

**Comment by sandewt** [comment]
@dautrich If you find the message (logon / logout) annoying, you can always make a language override.

**Comment by brianteeman** [comment]
> I'll wait what happens with this bug report. Hopefully a solution will solve the issue with SKYLAR as well. If not, I will open an issue in Ciaran's forum.

It will not be possible to change the code in the core to resolve your specific issue as it is with the template and not within any of the core code..

This should be closed as there is nothing in the core to be changed.

**Comment by brianteeman** [comment]
> @brianteeman I don't think that the issue should be closed. When you have a look at the original bug report, you see Cassiopeia, And the bug report is about the logged-in message being issued before the second part of the login, the 2FA, has been successfully done.

Sorry I was responding to the subsequent posts about the hornav and not the original one about the message.

For me the problem is that this "logged in message" should never have been merged as it is not _compatible_ with second factor authentication

**Comment by J-Wick4** [comment]
I agree with @brianteeman. Until you pass the MFA and are indeed logged into the system, posting a message you are logged in is misleading and confusing. Also, do we need to be told we are logged in? The only message I see as helpful is in the case of a wrong password or failed MFA attempt by giving a guidance message to try again, reset the password, etc.

**Comment by richard67** [comment]
Well the message is the symptom but not the problem. The problem is indeed that we are already logged in before we have passed MFA.

**Comment by dautrich** [comment]
When user email notifications are set, the "Login" notification also appears before the user has finished his MFA.

**Comment by brianteeman** [comment]
@SimonJ112 thats why you can decide which modules to display https://github.com/joomla/joomla-cms/issues/40428#issuecomment-1516063567
